modules/programs/spotify-player.nix

```
@@ -0,0 +1,154 @@
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib) mkEnableOption mkOption types literalExpression mkIf;
+
+  cfg = config.programs.spotify-player;
+  tomlFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = with lib.maintainers; [ diniamo ];
+
+  options.programs.spotify-player = {
+    enable = mkEnableOption "spotify-player";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.spotify-player;
+      defaultText = literalExpression "pkgs.spotify-player";
+      description = "The package to use for the spotify-player binary.";
+    };
+
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
+      example = literalExpression ''
+        {
+          theme = "default";
+          playback_window_position = "Top";
+          copy_command = {
+            command = "wl-copy";
+            args = [];
+          };
+          device = {
+            audio_cache = false;
+            normalization = false;
+          };
+        }
+      '';
+      description = ''
+        Configuration written to
+        {file}`$XDG_CONFIG_HOME/spotify-player/app.toml`.
+
+        See
+        <https://github.com/aome510/spotify-player/blob/master/docs/config.md#general>
+        for the full list of options.
+      '';
+    };
+
+    themes = mkOption {
+      type = types.listOf types.attrs;
```

not sure if this works, but I think it's better than just `types.attrs` if it works

```suggestion
      type = types.listOf tomlFormat.type;
```